src/git-version.cr

```diff
@@ -0,0 +1,141 @@
+require "file_utils"
+
+require "semantic_version"
+
+module GitVersion
+  extend self
+
+  BASE_VERISON_STRING = "0.0.0"
+  BASE_VERISON        = SemanticVersion.parse(BASE_VERISON_STRING)
+
+  DEV_BRANCH_SUFFIX = "SNAPSHOT"
+
+  MASTER_BRANCH = "master"
+
+  MAJOR_BUMP_COMMENT = "breaking:"
+  MINOR_BUMP_COMMENT = "feature:"
+
+  class Git
+    def initialize(@devBranch : String, @folder = FileUtils.pwd)
+      #
+    end
+
+    private def exec(cmd)
+      strout = IO::Memory.new
+
+      if !Process.run(
+           command: cmd,
+           shell: true,
+           output: strout,
+           error: Process::Redirect::Close,
+           chdir: @folder
+         ).success?
+        raise "[ERROR] Command #{cmd} failed."
+      end
+
+      return strout.to_s.split('\n', remove_empty: true)
+    end
+
+    def tags_by_branch(branch)
+      return exec "git tag --merged #{branch}"
```

Comment:
Does this work for when people rebase dev to master?
example:
`git checkout master && git rebase dev && git push`

Maybe a test for this would be cool

Reply:
added a specific test for the use case